generate-docs/scripts/preprocessor.ts

```diff
@@ -124,6 +109,53 @@ tryCatch(async () => {
         dtsBuilder.extractDtsSection(definitions, "Begin Word APIs", "End Word APIs")
     );
 
+    // ----
+    // Process Custom Functions d.ts
+    // ----
+    const dtsForCfs = "https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/custom-functions-runtime/index.d.ts";
+    fsx.writeFileSync("../script-inputs/custom-functions-runtime.d.ts", await fetchAndThrowOnError(dtsForCfs, "text"));
+    console.log(`\nReading from ${path.resolve("../script-inputs/custom-functions-runtime.d.ts")}`);
+    let definitionsForCfs : string = fsx.readFileSync("../script-inputs/custom-functions-runtime.d.ts").toString();
+
+    console.log("Fixing issues with d.ts file...");
+    definitionsForCfs = definitionsForCfs.replace(/^(\s*)(declare namespace)(\s+)/gm, `$1export $2$3`)
+        .replace(/^(\s*)(declare module)(\s+)/gm, `$1export $2$3`)
+        .replace(/^(\s*)(namespace)(\s+)/gm, `$1export $2$3`)
+        .replace(/^(\s*)(class)(\s+)/gm, `$1export $2$3`)
+        .replace(/^(\s*)(interface)(\s+)/gm, `$1export $2$3`)
+        .replace(/^(\s*)(module)(\s+)/gm, `$1export $2$3`)
+        .replace(/^(\s*)(function)(\s+)/gm, `$1export $2$3`)
+        .replace(/(\s*)(@param)(\s+)(\w+)(\s)(\s)/g, `$1$2$3$4$5`)
+        .replace(/(\s*)(@param)(\s+)(\w+)(\s+)([^\-])/g, `$1$2$3$4$5- $6`);
```

Comment:
Can we pull this regex into a separate function since it's used multiple times?

Reply:
Yep, good suggestion. I've moved 'common' (reused) regex logic into a function.  Please note that (by design) the Office-js d.ts processing logic still contains some regex logic for stuff that only needs done to that one file (i.e., stuff that isn't needed within Custom Functions logic or Office Runtime logic).